src/pybind/mgr/cephadm/module.py

```diff
@@ -1756,6 +1765,12 @@ def add_rgw(self, spec):
             'name': 'rgw_realm',
             'value': spec.rgw_realm,
         })
+        ret, out, err = self.mon_command({
+            'prefix': 'config set',
+            'who': 'client.rgw.' + spec.name,
+            'name': 'rgw_frontends',
+            'value': spec.rgw_frontend_config_value(),
+        })
```

Comment:
While testing this, I don't see RGW picking up this rgw_frontends config option. Is there something I'm missing in setting this up? I assumed the ConfigProxy would search the mon config store, is this not the case?

I guess I could debug to find out.

Reply:
Yeah, I think this is currently the major problem in the patch. As far as I can see, RGW doesn't look for the rgw_frontends value in the mon config store. I'd just assumed this was built into the the ConfigProxy.

Should we change @yehudasa 's https://github.com/ceph/ceph/pull/33287 to make the get keys from config-key available for other RGW values. And store the rgw_frontends there too (I guess check if it's missing in ceph.conf and  before defaulting to the default frontend configuration)?

ATM the only way to get this `rgw_frontends` working as it is, with out another RGW change is to write it into the ceph.conf in the container.. which I'd really rather not do.

Thoughts?

Reply:
But if that is correct (not search config store) then the rgw_realm and rgw_zone is also working above.

I wonder how hard it would be to add mon config support to ConfigProxy.. though I guess continual calls to the network could be problematic. Or does this actaully work and I've just missed something obvious?

Reply:
@matthewoliver there is a problem with rgw order of initialization that rgw_frontends happens before it even connects to the cluster (it seems like it). I'm not sure that changing it won't break something else. In any case I'm working on adding a default frontends configurable that will allow using the config-key the way @liewegas suggested.

Reply:
@matthewoliver my PR now added default ssl cert/key configuration for the beast frontend via config-key.

Reply:
Awesome! I'll rebase this on your PR on my dev machine and rework this one based on it. That way we can give them both a test.